atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbRuntimeConfig.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.config;
+
+import javax.annotation.Nullable;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.palantir.atlasdb.AtlasDbConstants;
+
+@JsonDeserialize(as = ImmutableAtlasDbRuntimeConfig.class)
+@JsonSerialize(as = ImmutableAtlasDbRuntimeConfig.class)
+@Value.Immutable
+public abstract class AtlasDbRuntimeConfig {
+    /**
+     * If true, a background thread will periodically delete cells that
+     * have been overwritten or deleted. This differs from scrubbing
+     * because it is an untargeted cleaning process that scans all data
+     * looking for cells to delete.
+     */
+    @Value.Default
+    public boolean enableSweep() {
+        return AtlasDbConstants.DEFAULT_ENABLE_SWEEP;
+    }
+
+    /**
+     * The number of milliseconds to wait between each batch of cells
+     * processed by the background sweeper.
+     */
+    @Value.Default
+    public long getSweepPauseMillis() {
+        return AtlasDbConstants.DEFAULT_SWEEP_PAUSE_MILLIS;
+    }
+
+    /**
+     * The target number of (cell, timestamp) pairs to examine in a single run of the background sweeper.
+     */
+    // TODO(gbonik): make this Default after we delete the deprecated options. For now, we need to be able to detect
```

Comment:
This is already breaking by adding a new config block, the `atlas-runtime`, which will be live-reloaded... I don't know if we want to not support getSweepBatchSize, getSweepCellBatchSize on the new config though. This would potentially make the upgrade to a new atlas version harder (even though not by a large margin).